### PR TITLE
iptables: Add override for iptables validation timeout

### PR DESCRIPTION
**Pull request by lambdai**
and reset default value to 5 seconds

The validation usually complete in 0.2 seconds. The false positive rate at 5 seconds is almost 0.
Well, also the validation doesn't really need super fast in case of validation failure.

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

### CODE REVIEW

File: tools/istio-iptables/pkg/cmd/root.go [hunk 1]
```diff
@@ -260,6 +261,12 @@ func init() {
 	}
 	viper.SetDefault(constants.IptablesProbePort, strconv.Itoa(constants.DefaultIptablesProbePort))
 
+	rootCmd.Flags().String(constants.ProbeTimeout, strconv.FormatUint(constants.DefaultProbeTimeout, 10), "set timeout in nanosecond for failure detection")
```
Comment (rlenglet) [inline tools/istio-iptables/pkg/cmd/root.go]:
```suggestion
	rootCmd.Flags().Duration(constants.ProbeTimeout, constants.DefaultProbeTimeout, "failure detection timeout")
```

File: tools/istio-iptables/pkg/config/config.go [hunk 2]
```diff
@@ -37,6 +36,7 @@ type Config struct {
 	OutboundIPRangesExclude string `json:"OUTBOUND_IPRANGES_EXCLUDE"`
 	KubevirtInterfaces      string `json:"KUBEVIRT_INTERFACES"`
 	IptablesProbePort       uint16 `json:"IPTABLES_PROBE_PORT"`
+	ProbeTimeout            uint64 `json: "PROBE_TIMEOUT"`
```
Comment (rlenglet) [inline tools/istio-iptables/pkg/config/config.go]:
```suggestion
	ProbeTimeout            time.Duration `json: "PROBE_TIMEOUT"`
```

File: tools/istio-iptables/pkg/constants/constants.go [hunk 3]
```diff
@@ -112,6 +115,7 @@ const (
 
 const (
 	DefaultIptablesProbePort = 15002
+	DefaultProbeTimeout      = uint64(5 * time.Second)
```
Comment (rlenglet) [inline tools/istio-iptables/pkg/constants/constants.go]:
```suggestion
	DefaultProbeTimeout      = 5 * time.Second
```

File: tools/istio-iptables/pkg/validation/validator.go [hunk 4]
```diff
@@ -118,6 +119,7 @@ func NewValidator(config *config.Config, hostIP net.IP) *Validator {
 			ServerOriginalPort:  config.IptablesProbePort,
 			ServerOriginalIP:    serverIP,
 			ServerReadyBarrier:  make(chan ReturnCode, 1),
+			ProbeTimeout:        time.Duration(config.ProbeTimeout),
```
Comment (rlenglet) [inline tools/istio-iptables/pkg/validation/validator.go]:
```suggestion
			ProbeTimeout:        config.ProbeTimeout,
```

File: tools/istio-iptables/pkg/cmd/root.go (hunk 1 above)
Comment (lambdai) [reply tools/istio-iptables/pkg/cmd/root.go]: TIL!

**Comment by lambdai** [comment]
@rlenglet Thank you for the detailed suggestions!

File: tools/istio-iptables/pkg/cmd/root.go (hunk 1 above)
Comment (rlenglet) [reply tools/istio-iptables/pkg/cmd/root.go]: FYI, this allows setting values like `1s`, `200ms`, etc. cf. https://golang.org/pkg/time/#ParseDuration

**Comment by lambdai** [comment]
/wait 
rebase after pickup

**Comment by lambdai** [comment]
cherrypick/release-1.4

**Pull request merged**